synapse/app/_base.py

```diff
@@ -161,7 +180,7 @@ def quit_with_error(error_string: str) -> NoReturn:
     sys.exit(1)
 
 
-def handle_startup_exception(e: Exception) -> NoReturn:
+def handle_startup_exception(e: Exception) -> None:
```

Comment:
I think this therefore ought to be `NoReturn` too. I don't know if mypy can/should enforce that.